vowpalwabbit/conditional_contextual_bandit.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <cstdint>
+#include <vector>
+
+#include "label_parser.h"
+#include "v_array.h"
+#include "action_score.h"
+#include "options.h"
+
+
+namespace LEARNER {
+  template<class T, class E> struct learner;
+  using base_learner = learner<char, char>;

```

Comment:
What is this?

Reply:
Forward declaration due to headers being included in a weird order